client/me/account/main.jsx

```diff
@@ -492,7 +502,8 @@ const Account = React.createClass( {
 						id="url"
 						name="url"
 						onFocus={ this.recordFocusEvent( 'Web Address Field' ) }
-						valueLink={ this.valueLink( 'user_URL' ) }
+						value={ this.getUserSetting( 'user_URL' ) || '' }
+						onChange={ this.updateUserSettingInput( 'user_URL' ) }
```

Comment:
> I think it's also very useful if the linter doesn't let you put too much complex code into a JSX attribute, and forces you to extract the logic to a separate method. Even if there is no performance benefit from that.

this just isn't the point of the linting rule though. it's specifically there to catch a performance and behavioral issue that's easy to overlook and hard to debug. creating new closures in the props of a React component can do weird things and that's true whether by using the arrow functions inline or by calling a function which returns a new function.

we need to remember that there is no sense of a value comparison of functions. _every_ time `render()` gets called here we create a new function closure and even if all other things are identical that won't shallow-compare to equality with the previous run. this is both a performance issue and can trigger re-renders and it's also a bug vector because any state created in the function gets lost as we abandon the old function invocation.